cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,301 @@
+  private static final String NAMESPACE_COL = "namespace";
+  private static final String GENERATION_COL = "generation";
+  private static final String OWNER_COL = "owner";
+  private static final String ID_COL = "id";
+  private static final String ARTIFACT_COL = "artifact";
+  private static final String NAME_COL = "name";
+  private static final String DESCRIPTION_COL = "description";
+  private static final String CREATED_COL = "createdTimeMillis";
+  private static final String UPDATED_COL = "updatedTimeMillis";
+  private static final String PIPELINE_COL = "pipeline";
+  private static final String REVISION_COL = "revision";
+  public static final StructuredTableSpecification TABLE_SPEC = new StructuredTableSpecification.Builder()
+    .withId(TABLE_ID)
+    .withFields(Fields.stringType(NAMESPACE_COL),
+                Fields.longType(GENERATION_COL),
+                Fields.stringType(OWNER_COL),
+                Fields.stringType(ID_COL),
+                Fields.stringType(ARTIFACT_COL),
+                Fields.stringType(NAME_COL),
+                Fields.stringType(DESCRIPTION_COL),
+                Fields.longType(CREATED_COL),
+                Fields.longType(UPDATED_COL),
+                Fields.stringType(PIPELINE_COL),
+                Fields.intType(REVISION_COL))
+    .withPrimaryKeys(NAMESPACE_COL, GENERATION_COL, OWNER_COL, ID_COL)
+    .build();
+  private static final Gson GSON = new Gson();
+  private final TransactionRunner transactionRunner;
+
+  public DraftStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  /**
+   * @param namespace the namespace to fetch the drafts from
+   * @param owner the id of the owner of the drafts
+   * @param sortRequest The sorting that should be applied to the results, pass null if no sorting is required.
+   * @return a list of drafts
+   * @throws TableNotFoundException if the draft store table is not found
+   * @throws InvalidFieldException if the fields Namespace and owner fields do not match the fields in the
+   *   StructuredTable
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespace, String owner,
+                                SortRequest sortRequest) throws TableNotFoundException, InvalidFieldException {
+    List<Field<?>> prefix = new ArrayList<>(3);
+    prefix.add(Fields.stringField(NAMESPACE_COL, namespace.getName()));
+    prefix.add(Fields.longField(GENERATION_COL, namespace.getGeneration()));
+    prefix.add(Fields.stringField(OWNER_COL, owner));
+
+    List<StructuredRow> rows;
+    rows = TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      Range range = Range.singleton(prefix);
+      List<StructuredRow> temp = new ArrayList<>();
+      try (CloseableIterator<StructuredRow> rowIter = table.scan(range, Integer.MAX_VALUE)) {
+        while (rowIter.hasNext()) {
```

Comment:
You can simplify it to `rowIter.forEachRemaining(temp::add)`

Reply:
done